sdks/python/apache_beam/coders/coder_impl.py

```diff
@@ -278,38 +277,38 @@ def get_estimated_size_and_observables(self, value, nested=False):
     return out.get_count(), []
 
   def encode_to_stream(self, value, stream, nested):
-    t = type(value)
-    if t is NoneType:
+    if value is None:
       stream.write_byte(NONE_TYPE)
```

Comment:
<!--new_thread; commit:05e95b6995f0b88a53930c30459c8d81e6692219; resolved:0-->
Can you revert the change removing `t = type(value)`?  I believe this is done because `isinstance` is much slower than `is`, and this is very performance-sensitive code.

CC: @robertwb

Reply:
This was changed because `is` doesn't check for subclasses. The whole future.builtins package depends on subclasses to add compatibility.
This can however be changed by depending on six again if necessary.

Reply:
I have just checked the timing myself. 
`type(list()) is list` is exactly as fast as `isinstance(list(), list)`, but with an average of 5 if checks for every type call, the type check is 2.5x faster.